modules/optflow/src/rlof/rlof_invokerbase.hpp

```diff
@@ -28,8 +27,88 @@ using namespace cv;
 namespace cv {
 namespace optflow {
 
+
 typedef short deriv_type;
 #ifdef RLOF_SSE
+
+#if CV_SSE4_1 | CV_SSE4_2
+#define ABS_EPI16(a) _mm_abs_epi16(a)
+#define CVTEPI8_EPI16(a) _mm_cvtepi8_epi16(a)
+#define CVTEPI16_EPI32(a) _mm_cvtepi16_epi32(a)
+#define BLENDV_EPI8(a, b, mask) _mm_blendv_epi8(a, b, mask)
+#define BLENDV_PS(a, b, mask) _mm_blendv_ps(a, b, mask)
+#endif
+
+#ifndef ABS_EPI16
+static inline __m128i cabs_epi16(__m128i _a)
+{
+    short CV_DECL_ALIGNED(16) a[8];
+    _mm_storeu_si128((__m128i*)a, _a);
+    return _mm_set_epi16(
+        static_cast<short>(std::abs(a[7])), static_cast<short>(std::abs(a[6])),
+        static_cast<short>(std::abs(a[5])), static_cast<short>(std::abs(a[4])),
+        static_cast<short>(std::abs(a[3])), static_cast<short>(std::abs(a[2])),
+        static_cast<short>(std::abs(a[1])), static_cast<short>(std::abs(a[0])));
+}
+#define ABS_EPI16(a) cabs_epi16(a)
+#endif
+#ifndef CVTEPI8_EPI16
+static inline __m128i cvtepi8_epi16(__m128i _a)
+{
+    char CV_DECL_ALIGNED(8) a[64];
+    _mm_storeu_si128((__m128i*)a, _a);
+    return _mm_set_epi16(static_cast<int>(a[7]), static_cast<int>(a[6]),
+                         static_cast<int>(a[5]), static_cast<int>(a[4]),
+                         static_cast<int>(a[3]), static_cast<int>(a[2]),
+                         static_cast<int>(a[1]), static_cast<int>(a[0]));
+}
+#define CVTEPI8_EPI16(a) cvtepi8_epi16(a)
+#endif
+#ifndef CVTEPI16_EPI32
+static inline __m128i cvtepi16_epi32(__m128i _a)
+{
+   short CV_DECL_ALIGNED(16) a[8];
+   _mm_storeu_si128((__m128i*)a, _a);
+   return _mm_set_epi32(static_cast<int>(a[3]), static_cast<int>(a[2]), static_cast<int>(a[1]), static_cast<int>(a[0]));
+}
+#define CVTEPI16_EPI32(a) cvtepi16_epi32(a)
+#endif
+#ifndef BLENDV_EPI8
+static inline __m128i blendv_epi8(__m128i _a, __m128i _b, __m128i _mask)
+{
+    short CV_DECL_ALIGNED(16) a[8], b[8], res[8];
+    unsigned short CV_DECL_ALIGNED(16) mask[8];
+    _mm_storeu_si128((__m128i*)a, _a);
+    _mm_storeu_si128((__m128i*)b, _b);
+    _mm_storeu_si128((__m128i*)mask, _mask);
+    res[0] = (mask[0] > 0) ? b[0] : a[0];
+    res[1] = (mask[1] > 0) ? b[1] : a[1];
+    res[2] = (mask[2] > 0) ? b[2] : a[2];
+    res[3] = (mask[3] > 0) ? b[3] : a[3];
```

Comment:
This code doesn't look optimal.
Take a look on OpenCV HAL intrinsic (SIMD) implementation:
- SSEx [v_select](https://github.com/opencv/opencv/blame/4.2.0/modules/core/include/opencv2/core/hal/intrin_sse.hpp#L446-L454)
- SSEx [v_abs for int16x8](https://github.com/opencv/opencv/blame/4.2.0/modules/core/include/opencv2/core/hal/intrin_sse.hpp#L1042-L1047)

It would be nice if you can port this raw SSE code to [OpenCV HAL intrinsics](https://docs.opencv.org/3.4/df/d91/group__core__hal__intrin.html#details) - this code would run on ARM NEON automatically.